drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/housing_assessment_importer.rb

```diff
@@ -236,17 +236,30 @@ def ensure_intake_assessment!(waitlist, enrollment)
     ActiveRecord::Base.record_timestamps = true
   end
 
-  # TODO: if not found by MCI Unique ID, should we try to look up by MCI ID?
   def find_and_update_hmis_client(waitlist)
     # client_id is the MCI Unique ID
     mci_scope = HmisExternalApis::ExternalId.
       where(namespace: HmisExternalApis::AcHmis::WarehouseChangesJob::NAMESPACE).
       where(value: waitlist.client_id)
 
     client = Hmis::Hud::Client.joins(:ac_hmis_mci_unique_id).merge(mci_scope).first
-    return nil unless client
+    if client
+      log_info("Found client with MCI Unique ID #{waitlist.client_id}: #{client.id}")
+      return client
+    end
+
+    # If not found by MCI Unique ID, look up by MCI ID. This would be needed if the client exists in HMIS
+    # because they were referred from LINK, but they don't have an MCI Unique ID. When Link sends a referral
+    # "posting" for a new client, we generate a new Client record with an MCI ID. We don't create an MCI Unique ID at that time,
+    # and the MCI Unique ID won't get created until/unless the client has any enrollments. (Unenrolled clients
+    # are not exported in HMIS export, so Data Warehouse API won't provide MCI Unique IDs for those clients.)
+    found_mci_ids = HmisExternalApis::ExternalId.mci_ids.where(value: waitlist.client_mci_id)
```

Comment:
nit: calling size and then first will yield two queries
```suggestion
    found_mci_ids = HmisExternalApis::ExternalId.mci_ids.where(value: waitlist.client_mci_id).to_a
```